### PR TITLE
feat(observability): otel logs and metrics

### PR DESCRIPTION
**Pull request by nferch**
Experimental support for OpenTelemetry Logs, Metrics, and Traces submitted through an OpenTelemetry collector.

* Both Datadog and OpenTelemetry metrics are submitted for now.
* Log capture must be explicitly started, and is not enabled on the CLI, probably need some better way for adjusting features.

TODO:
* Remove Datadog
* Update `@baselime/node-opentelemetry` to fix peer dependency issues (https://github.com/baselime/node-opentelemetry/pull/7)

### CODE REVIEW

File: packages/sdk/observability/src/otel/metrics.ts
Comment (wittjosiah) [inline packages/sdk/observability/src/otel/metrics.ts:1]: what do you think about hooking these metrics up to this interface? https://github.com/dxos/dxos/blob/main/packages/sdk/observability/src/sentry/browser.ts#L61 then any metrics that get added within the sdk will get sent to sentry for the apps and otel for agents.

Comment (nferch) [reply packages/sdk/observability/src/otel/metrics.ts:1]: the API seems similar enough, so I don't see why not.

were you thinking we shouldn't send metrics for the apps to Grafana?

could also look at sending traces to Grafana, but I'm less familiar with how we're using those.

Comment (wittjosiah) [reply packages/sdk/observability/src/otel/metrics.ts:1]: > were you thinking we shouldn't send metrics for the apps to Grafana?

not necessarily, but let's discuss

File: packages/sdk/observability/src/otel/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    });
+
+    this._meterProvider = new MeterProvider({
+      resource,
+      readers: [grafanaMetricReader],
+    });
+    this._meter = this._meterProvider.getMeter('dxos-observability');
+
+    const metrics = {
+      // TODO: update metrics names and remove prefix?
```
Comment (nferch) [inline packages/sdk/observability/src/otel/metrics.ts:63]: @wittjosiah @yarolegovich could we prefix the metric names (e.g. `dxos.echo.space.count`) so they're less ambiguous and consistent with the trace names?

**Pull request merged**